### ISSUE TITLE
Code Complexity

### ISSUE THREAD

**Issue by MahzaibKhalid** [Cuixxx/ProFD#3]
Hi, I need specific code from your project. But its is to complex to understand at first. Can you provide exact code for the paper?

**Comment by Cuixxx** [comment]
Sure, most of related parts are written in the file, './torchreid/models/profd.py'. And if you have more problems, please let me know! Thank you!

**Comment by MahzaibKhalid** [comment]
Actually, I am confused only about textual prompts with part based. I have complete understanding about rest. Can you provide me a template of only textual part based prompt as an example. It will make clarity about your textual prompts pattern. Thanks

**Comment by Cuixxx** [comment]
I use learnable prompt as template, e.g. [p0],[p1],...,[p15],[feet/arm/head/torso]. Notably, [p0],...,[p15] are all trainable embeddings and they are not pre-defined before training. I hope this answer can help you!

**Comment by MahzaibKhalid** [comment]
Ok got it. It means that you dynamically generated part specific prompts and add token of class at the end. Right?
![Screenshot 2024-10-18 154051](https://github.com/user-attachments/assets/8285710a-51fc-40a7-bddb-2b89e25a1ba3)

**Comment by Cuixxx** [comment]
Yes, you are right. If you have any other problems, please let me know.

**Comment by MahzaibKhalid** [comment]
I have one more question, You have generated dynamic prompts. How is the surety that these prompts are specific for that specific part? Is there any way to optimize the promots?

**Comment by Cuixxx** [comment]
Good Question! 
(1)The part-specific prompts means that  learnable template + body part name, I think this way mainly build their relationship between prompt and its corresponding body region. (2) Besides, we also propose a spatial-level alignment to further align their embedding with spatial feature map in spatial level, similar with a segmentation task, these prompt embedding can be regarded as a classifier head. After these two procedures, we believe these prompt embedding have well aligned with their corresponding body regions.
